### PR TITLE
Restore `clone-branch`

### PR DESCRIPTION
**Pull request by yakov116**
Github changed the selectors

Test on:
https://github.com/sindresorhus/refined-github/branches
https://github.com/yakov116/TestR/branches

Cool issue #3000
<!--

Thanks for contributing! 🍄

1. LINKED ISSUES:
   Does this PR close/fix an existing issue? Write something like `Closes #10`

2. TEST URLS:
   Add some test URLs

3. SCREENSHOT:
   Add a screenshot here if your PR makes visual changes

### CODE REVIEW

File: source/features/clone-branch.tsx [hunk 1]
```diff
@@ -37,9 +37,9 @@ async function createBranch(newBranchName: string, baseSha: string): Promise<tru
 
 async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>): Promise<void> {
 	const cloneButton = event.delegateTarget;
-	const branchElement = cloneButton.closest<HTMLElement>('[data-branch-name]')!;
+	const branchName = select('.branch-name', cloneButton.offsetParent!)!.textContent!;
```
Comment (fregante) [inline source/features/clone-branch.tsx:40]:
```suggestion
	const branchName = cloneButton.closest('[branch]')!.getAttribute('branch')!;
```

File: source/features/clone-branch.tsx [hunk 2]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
 	}
 
 	textFieldEdit.set(
-		select<HTMLInputElement>('.js-branch-search-field')!,
+		select<HTMLInputElement>('[aria-label="Search branches…"]')!,
```
Comment (fregante) [inline source/features/clone-branch.tsx:65]: Let’s use its public `name`, that shouldn't ever change
```suggestion
		select<HTMLInputElement>('[name="query"]')!,
```

File: source/features/clone-branch.tsx [hunk 3]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
 		newBranchName
 	);
 }
 
 function init(): void | false {
-	const deleteIcons = select.all('[aria-label="Delete this branch"]');
-	// Is the user does not have rights to create a branch
+	const deleteIcons = select.all('.Details-content--shown .octicon.octicon-trashcan');
```
Comment (fregante) [inline source/features/clone-branch.tsx:71]: This lets us skip the `.parentElement` below

```suggestion
	const deletionForms = select.all('form[data-action$="#destroy"]');
```

File: source/features/clone-branch.tsx [hunk 4]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
+	// If the user does not have rights to delete a branch
 	if (deleteIcons.length === 0) {
 		return false;
 	}
 
 	for (const branch of deleteIcons) {
-		branch.closest('.js-branch-destroy')!.before(
+		branch.parentElement!.before(
```
Comment (fregante) [inline source/features/clone-branch.tsx:78]:
```suggestion
		deletionForm.before(
```

File: source/features/clone-branch.tsx [hunk 5]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
 function init(): void | false {
-	const deleteIcons = select.all('[aria-label="Delete this branch"]');
-	// Is the user does not have rights to create a branch
+	const deleteIcons = select.all('.Details-content--shown .octicon.octicon-trashcan');
+	// If the user does not have rights to delete a branch
 	if (deleteIcons.length === 0) {
 		return false;
 	}
 
 	for (const branch of deleteIcons) {
```
Comment (fregante) [inline source/features/clone-branch.tsx:77]:
```suggestion
	for (const deletionForm of deletionForms) {
```

File: source/features/clone-branch.tsx [hunk 6]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
 		newBranchName
 	);
 }
 
 function init(): void | false {
-	const deleteIcons = select.all('[aria-label="Delete this branch"]');
-	// Is the user does not have rights to create a branch
+	const deleteIcons = select.all('.Details-content--shown .octicon.octicon-trashcan');
+	// If the user does not have rights to delete a branch
 	if (deleteIcons.length === 0) {
```
Comment (fregante) [inline source/features/clone-branch.tsx:73]:
```suggestion
	if (deletionForms.length === 0) {
```

File: source/features/clone-branch.tsx [hunk 7]
```diff
@@ -62,20 +62,20 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
+		select<HTMLInputElement>('[aria-label="Search branches…"]')!,
 		newBranchName
 	);
 }
 
 function init(): void | false {
-	const deleteIcons = select.all('[aria-label="Delete this branch"]');
-	// Is the user does not have rights to create a branch
+	const deleteIcons = select.all('.Details-content--shown .octicon.octicon-trashcan');
+	// If the user does not have rights to delete a branch
```
Comment (fregante) [inline source/features/clone-branch.tsx:72]:
```suggestion
	// If the user does not have rights to delete a branch, they can’t create one either
```

**Comment by fregante** [comment]
Congrats on opening discussion #3000 🎉

**Comment by yakov116** [comment]
> Congrats on opening discussion #3000 🎉

And i thought it would be simple 😄

File: source/features/clone-branch.tsx (hunk 3 above)
Comment (yakov116) [reply source/features/clone-branch.tsx]: I did that since 
image

Comment (yakov116) [reply source/features/clone-branch.tsx]: Found another way

File: source/features/clone-branch.tsx [hunk 8]
```diff
@@ -62,20 +62,23 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
 }
 
 function init(): void | false {
-	const deleteIcons = select.all('[aria-label="Delete this branch"]');
-	// Is the user does not have rights to create a branch
-	if (deleteIcons.length === 0) {
+	const deletionForms = select.all([
+		'form[data-action$="#destroy"]',
+		'[aria-label="You can’t delete this branch because it has an open pull request"]' // There is no form when there is an open pull request
+	]);
```
Comment (fregante) [inline source/features/clone-branch.tsx:74]:
```suggestion
	const deleteIcons = select.all('branch-filter-item-controller .octicon-trashcan');
```

File: source/features/clone-branch.tsx [hunk 9]
```diff
@@ -62,20 +62,23 @@ async function cloneBranch(event: delegate.Event<MouseEvent, HTMLButtonElement>)
...
+	// If the user does not have rights to delete a branch, they can’t create one either
+	if (deletionForms.length === 0) {
 		return false;
 	}
 
-	for (const branch of deleteIcons) {
-		branch.closest('.js-branch-destroy')!.before(
+	for (const deletionForm of deletionForms) {
+		deletionForm.before(
```
Comment (fregante) [inline source/features/clone-branch.tsx:81]:
```suggestion
	for (const deleteIcon of deleteIcons) {
		// Branches with open PRs use `span`; the others use `form`
		deleteIcon.closest('form, span')!.before(
```

**Pull request merged**